openadmet/models/features/gat_featurizer.py

```diff
@@ -0,0 +1,175 @@
+            ]
+            atom_features_list.append(features)
+
+        if not atom_features_list: # Should not happen if MolFromSmiles was successful and molecule has atoms
+            logger.warning(f"No atoms found for SMILES: {smiles} (mol object: {mol})")
+            return None
+
+        x = torch.tensor(atom_features_list, dtype=torch.float)
+
+        edge_indices = []
+        edge_features_list = []
+        for bond in mol.GetBonds():
+            start_idx = bond.GetBeginAtomIdx()
+            end_idx = bond.GetEndAtomIdx()
+
+            edge_indices.extend([[start_idx, end_idx], [end_idx, start_idx]])
+
+            bond_f = [
+                float(bond.GetBondTypeAsDouble()),
+                float(bond.GetIsAromatic()),
+                float(bond.IsInRing()),
+                float(bond.GetStereo()) # RDKit returns BondStereo, convert to float
+            ]
+            edge_features_list.extend([bond_f, bond_f])
+
+        if edge_indices:
+            edge_index = torch.tensor(edge_indices, dtype=torch.long).t().contiguous()
+            edge_attr = torch.tensor(edge_features_list, dtype=torch.float) if edge_features_list else None
+        else: # Handle molecules with a single atom (no bonds)
+            edge_index = torch.empty((2, 0), dtype=torch.long)
+            edge_attr = None # No edges, so no edge attributes
+
+        data = Data(x=x, edge_index=edge_index, edge_attr=edge_attr)
+
+        if y_val is not None:
+            data.y = torch.tensor([y_val], dtype=torch.float)
+
+        return data
+
+    def featurize(self, smiles_list: list[str], y_list: Optional[list[Any]] = None, batch_size: int = 32, shuffle: bool = False, num_workers: int = 0):
+        """
+        Featurize a list of SMILES strings into a PyTorch Geometric DataLoader.
+
+        Args:
+            smiles_list: List of SMILES strings.
+            y_list: Optional list of target values. Values will be attempted to be cast to float.
+            batch_size: Batch size for DataLoader.
+            shuffle: Whether to shuffle the data in DataLoader.
+            num_workers: Number of worker processes for DataLoader.
+
+        Returns:
+            Tuple of (DataLoader, None, List[Data]). The DataLoader for training,
+            scaler (None for graphs), and the list of Data objects.
+            Invalid SMILES or problematic molecules will be skipped (a warning will be logged).
+        """
+        if Chem is None:
+            logger.error("RDKit not available. Cannot featurize SMILES.")
+            return DataLoader([]), None
+
+        data_objects = []
+        # Convert y_list to a list if it's a pandas Series to avoid indexing issues
+        if y_list is not None:
+            if hasattr(y_list, 'tolist'):
+                y_list = y_list.tolist()
+            elif hasattr(y_list, 'values'):
+                y_list = y_list.values.tolist()
+
+        for i, smiles_str in enumerate(smiles_list):
```

Comment:
I wouldn't bother with this, expect people to put in numerical values or thrown an error.